### PR TITLE
Resolves #169: Remove deprecated KeySpacePath APIs

### LINKED ISSUE #169: Remove deprecated KeySpacePath API's
As part of issue #161, a number of API's have been deprecated.  These should be removed after the next major release.

### PR DESCRIPTION
**Pull request by scgray**

### CODE REVIEW

**Comment by alecgrieser** [comment]
It looks like a reference to one of the deprecated methods needs to be removed from a Javadoc comment based on the CI output:

```
/code/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/PathValue.java:30: error: reference not found
 * is the object that will appear in the {@link com.apple.foundationdb.tuple.Tuple} when {@link KeySpacePath#toTuple()}
                                                                                                ^
/code/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/PathValue.java:35: warning - Tag @link: can&#39;t find toTuple() in com.apple.foundationdb.record.provider.foundationdb.keyspace.KeySpacePath
```

**Comment by alecgrieser** [comment]
Probably needs a rebase against fdb-record-layer-develop-2.6 now that that exists.

**Comment by scgray** [comment]
@alecgrieser done.

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ As the [versioning guide](Versioning.md) details, it cannot always be determined
 * **Feature** Feature 3 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Feature** Feature 4 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Feature** Feature 5 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
-* **Breaking change** Change 1 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
+* **Breaking change** Remove deprecated KeySpacePath APIs [(Issue #169)](https://github.com/FoundationDB/fdb-record-layer/issues/169)
```
Comment (alecgrieser) [inline docs/ReleaseNotes.md]:
```suggestion
* **Breaking change** Remove deprecated `KeySpacePath` APIs [(Issue #169)](https://github.com/FoundationDB/fdb-record-layer/issues/169)
```

**Pull request merged**